doc/model.rst

```diff
@@ -0,0 +1,683 @@
+*****
+Model
+*****
+========
+Overview
+========
+SmartSim Models are an abstract representation for compiled applications,
+scripts, and general computational tasks. Models can be launched with
+other SmartSim entities and ML infrastructure to build AI-enabled workflows.
+SmartSim enables AI integration in ``Model`` workflows by allowing users to dynamically load
+TensorFlow (TF) functions, TorchScript, and various machine learning models
+(TF, TF-lite, PyTorch, or ONNX) into the database at runtime.
+Models are flexible enough to support many different applications, however,
+to be used with SmartSim clients (SmartRedis) the application must be written
+in Python, C, C++, or Fortran.
+
+When initializing a ``Model``, users
+provide executable simulation code to the Models run settings as well as
+execution instructions with regard to the workload
+manager (e.g. Slurm) and available compute resources. Users can specify
```

Comment:
The discussion of parameters feels confusing. I think there is a non-obvious segue from parameterized to configured model initialization tripping me up.

Consider introducing the two mechanisms first, e.g.

> Model parameters can be supplied directly during instantiation or through configuration files. The :ref:`Experiment API<experiment_api>` supports passing parameters up front. Dynamically loading configuration at runtime is enabled by the :ref:`Model API<model_api.attach_generator_files>`

With that blurb injected before "Users can specify..." I think I'd actually prefer the details about the specific argument names moved into a section outside the overview but defer to you.

Reply:
I agree with you, what I can do is link the reader to the section with the actual blurb info on params and such

Reply:
Im not sure if Im going to even include the attach_generator_file function tbh

Reply:
I think the Overview Im gonna try to keep as just an overview and not specific as before